andela/dahlia-ah-backend
# #167164983 Send Password Reset Email

#### Pivotal tracker story
[#167164983](https://www.pivotaltracker.com/story/show/167164983)

#### What does this PR do?
Sends a mail to the user with the password reset link

#### Summary of Task
  - create api key on sendGrid
  - create email sending module
  - construct appropiate email message
  - write unit tests
  - create user model
  - create endpoint

#### How can this be tested?
  - clone this branch to your local repository
  - set the required environment variables from the .env.sample file
  - run npm test for automated test
  - to test on Postman:
         - add a new user object to the seeder file to mimick a registered user
           and ensure you use an email which you can check for the link
         - start the server using `npm run dev:start`
         - go to the `/api/v1/auth/forgotpassword` endpoint
            and supply the email of the user you put in the seeder file

#### Screenshots (if appropriate)
N/A
